lib/nx_zlib.h

```diff
@@ -336,6 +338,45 @@ typedef struct nx_stream_s {
 } nx_stream;
 typedef struct nx_stream_s *nx_streamp;
 
+/* metric used to decide if it's time to switch to sw, as we are not able
+   to switch mid-stream we cannot use avg_delay directly.  */
```

Comment:
I didn't understand this comment. I'm afraid I'm not understanding because I can't distinguish how `metric` and `avg_delay` are being used.
What prevents `metric` from being updated after every `nx_submit_job()`?

Reply:
`metric` is used for the comparison in the `use_nx_*` functions, if we update it on `nx_submit_job()` and it passes the threshold it would try to switch to software in the middle of a NX stream and it would fail.

Reply:
Then I'm afraid we have an issue here because other threads may update `metric` causing a thread to try to switch in the middle of the execution of a stream.

Should `use_nx_*` only be called by `*Init` functions?